### PR TITLE
submission update fix

### PR DESCRIPTION
**Pull request by 0xf8f8ff**
Fix for new submissions not being updated correctly in the database.

### CODE REVIEW

**Comment by meling** [comment]
The change looks fine on its own, but I'll merge into master once we have revised tests for this.

File: database/gormdb.go [hunk 1]
```diff
@@ -397,7 +397,9 @@ func (db *GormDB) CreateSubmission(submission *pb.Submission) error {
 
 	// If a submission for the given assignment and student/group already exists, update it.
 	// Otherwise create a new submission record
-	return db.conn.Where(query).Assign(submission).FirstOrCreate(submission, query).Error
+	var labSubmission pb.Submission
+	return db.conn.Where(query).Assign(submission).FirstOrCreate(&labSubmission).Error
+
```
Comment (meling) [inline database/gormdb.go]: Remove empty line

**Comment by meling** [comment]
On the other hand, I see that the test that no longer passes seems to be the test for `GetNextAssignment` which may be deprecated going forward, since we now base our assignment selection on the actual push event (what changed)... However, the changed method should tested with data that passed before, but no longer passes... (Didn't analyze the details of the tests that exists related to `CreateSubmission`...)

File: database/gormdb.go (hunk 1 above)
Comment (0xf8f8ff) [reply database/gormdb.go]: Fixed.

**Comment by 0xf8f8ff** [comment]
The test was failing due to `FirstOrCreate` not affecting the original structure any more. All submissions in the test had ID=0 because of that. 

Got it working for now, will replace with simpler test when this one becomes deprecated.

**Pull request merged**